### PR TITLE
CHET-388: Initial mocked-server Cypress testing

### PR DESCRIPTION
**Pull request by tarka**

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -38,6 +38,14 @@
             }
         };
     </script>
+
+    <!-- Testing hack; Cypress can't mock fetch to force the installation of a polyfill that falls-back to XHR. -->
```
Comment (bpar476) [inline frontend/public/index.html:42]: Is fetch available when running via amps?

File: frontend/public/unfetch-polyfill.js [hunk 2]
```diff
@@ -0,0 +1 @@
+self.fetch||(self.fetch=function(e,n){return n=n||{},new Promise(function(t,s){var r=new XMLHttpRequest,o=[],u=[],i={},a=function(){return{ok:2==(r.status/100|0),statusText:r.statusText,status:r.status,url:r.responseURL,text:function(){return Promise.resolve(r.responseText)},json:function(){return Promise.resolve(JSON.parse(r.responseText))},blob:function(){return Promise.resolve(new Blob([r.response]))},clone:a,headers:{keys:function(){return o},entries:function(){return u},get:function(e){return i[e.toLowerCase()]},has:function(e){return e.toLowerCase()in i}}}};for(var c in r.open(n.method||"get",e,!0),r.onload=function(){r.getAllResponseHeaders().replace(/^(.*?):[^\S\n]*([\s\S]*?)$/gm,function(e,n,t){o.push(n=n.toLowerCase()),u.push([n,t]),i[n]=i[n]?i[n]+","+t:t}),t(a())},r.onerror=s,r.withCredentials="include"==n.credentials,n.headers)r.setRequestHeader(c,n.headers[c]);r.send(n.body||null)})});
```
Comment (bpar476) [inline frontend/public/unfetch-polyfill.js:1]: Is this some open-source polyfill? Do we need to include an acknowledgement or a reference?

File: jira-e2e-tests/cypress.json [hunk 3]
```diff
@@ -6,5 +6,7 @@
     "screenshotsFolder": "screenshots",
     "supportFile": "support/index.js",
     "testFiles": "**/*.*",
-    "videosFolder": "videos"
+    "videosFolder": "videos",
```
Comment (bpar476) [inline jira-e2e-tests/cypress.json:9]: Should we add these folders to the gitignore?

File: jira-e2e-tests/integration/amps_mocked.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+describe('Database Migration page', () => {
+    const ctx = jira.amps_context;
+
+    beforeEach(() => {
+        cy.on('uncaught:exception', (err, runnable) => false);
+
+        // Delete `fetch()` to force use of XHR for
+        // Cypress. Atlaskit/Jira will polyfill this.
+        Cypress.on('window:before:load', (win) => {
+            delete win.fetch
```
Comment (bpar476) [inline jira-e2e-tests/integration/amps_mocked.ts:14]: How it the polyfill loaded in this scenario? 🤔

File: frontend/public/index.html (hunk 1 above)
Comment (tarka) [reply frontend/public/index.html:42]: Yep, there's a polyfill available. I assume Atlaskit/Jira pull it in at some point for legacy browsers.

File: frontend/public/unfetch-polyfill.js (hunk 2 above)
Comment (tarka) [reply frontend/public/unfetch-polyfill.js:1]: Good point. I'll check. We can always source it directly.

Comment (tarka) [reply frontend/public/unfetch-polyfill.js:1]: It's MIT; I'll just source from a CDN to bypass the whole question.

File: jira-e2e-tests/cypress.json (hunk 3 above)
Comment (tarka) [reply jira-e2e-tests/cypress.json:9]: They're in `jira-e2e-tests/.gitignore` already.

File: jira-e2e-tests/integration/amps_mocked.ts (hunk 4 above)
Comment (tarka) [reply jira-e2e-tests/integration/amps_mocked.ts:14]: Already in Jira/Atlaskit/

**Pull request merged**